libsoftwarecontainer/src/gateway/networkgateway.cpp

```diff
@@ -24,17 +24,25 @@
 #include "networkgateway.h"
 #include "networkgatewayparser.h"
 
-NetworkGateway::NetworkGateway(const int32_t id) :
+NetworkGateway::NetworkGateway(const int32_t id,
+                               const std::string gateway,
+                               const uint8_t maskBits) :
     Gateway(ID),
-    m_netmask(0xFF),
-    m_gateway("10.0.3.1"),
+    m_gateway(gateway),
     m_interfaceInitialized(false),
-    m_containerID(id) { }
-
-NetworkGateway::~NetworkGateway()
+    m_containerID(id)
 {
+    if (maskBits > 31 || maskBits < 1) {
+        log_error() << "inappropriate netmask : " << maskBits;
+        m_netmask = 0xFF;
+        return;
+    }
+
+    m_netmask = (1L<<(32-maskBits));
```

Comment:
I think you should write the magic numbers in hex or binary when dealing with bits